src/index.js

```diff
@@ -1,115 +1,326 @@
+        console.log();
+        log(chalk`🍳  Cooking up a fresh new app...`);
         console.log();
 
-        await run(`${useYarn ? 'yarn add' : 'npm install'} ${devPackages.join(' ')} --${useYarn ? 'dev' : 'save-dev'}`, {
-            cwd: path.join(process.cwd(), answers.appname),
-        });
-    }
+        // run question execs
+        await this.runQuestionExecs();
+    };
+
+    runQuestionExecs = async () => {
+        for (const answerKey of Object.keys(this.answers)) {
+            const answer = this.answers[answerKey];
+            if (answerKey !== QUESTION_TYPES.appname) {
+                const Exec = questions[answerKey].execute;
+                const state = store.getState();
+                const params = {
+                    answers: this.answers,
+                    packages: this.packages,
+                    devPackages: this.devPackages,
+                    answer: answer,
+                    state,
+                };
+                const currentGenerator = state.generator;
+                const instance = new Exec(params);
+
+                // onPreInstall, handy if you need to set up things and want to use lifecycle things (you could also just use the constructor, but the option is nice to have.
+                if (typeof instance.onPreInstall === 'function') {
+                    await instance.onPreInstall();
+                }
+
+                // Push onPostInstall if function to functions array to be run, well, post install.
+                if (typeof instance.onPostInstall === 'function') {
+                    this.postInstallFuncs.push(instance.onPostInstall);
+                }
+
+                // If no answer was given or answer was false (this is handy when you want to i.e remove certain webTemplate strings.
+                if ((typeof answerKey === 'boolean' && answerKey === false) && !!instance.onNoAnswer) {
+                    await instance.onNoAnswer();
+                    continue;
+                }
+
+                // If using default function or something else is wrong
+                if (!instance[currentGenerator]) {
+                    if (typeof instance.default === 'function') {
+                        // Run default method
+                        await instance.default();
 
-    log('🚚  All packages installed.');
+                        // Retrieve packages that might be modified and set them to vars
+                        const { packages: packagesNew, devPackages: devPackagesNew } = instance.getAllPackages;
+                        this.packages = packagesNew;
+                        this.devPackages = devPackagesNew;
 
-    for (const key of Object.keys(questions)) {
-        if (key in postInstall) {
-            await postInstall[key]({ answer: answers[key], answers });
+                        // Next question.
+                        continue;
+                    } else {
+                        if (process.env.DEBUG === 1) {
```

Comment:
logs met type debug worden zoiezo alleen gelogged als process.env.DEBUG 1 is (zie `services/log.js`)
Daarnaast zijn alle values in `process.env` allemaal strings dus dit zal nooit naar true evaluaten